File: app/src/main/java/com/app/screensharinggrypp/AppLifecycleObserver.java
package com.app.screensharinggrypp;

import androidx.lifecycle.DefaultLifecycleObserver;
import androidx.lifecycle.LifecycleOwner;

public class AppLifecycleObserver implements DefaultLifecycleObserver {

    private final GlobalActionBarService globalService;

    public AppLifecycleObserver(GlobalActionBarService service) {
        this.globalService = service;
    }

    @Override
    public void onStop(LifecycleOwner owner) {
        if (globalService != null) {
            globalService.hideOverlayButton();
        }
    }

    @Override
    public void onStart(LifecycleOwner owner) {
        if (globalService != null) {
            globalService.showOverlayButton();
        }
    }
}

File: app/src/main/java/com/app/screensharinggrypp/AppLifecycleObserver.kt
package com.app.screensharinggrypp;

import androidx.lifecycle.DefaultLifecycleObserver
import androidx.lifecycle.LifecycleOwner


class AppLifecycleObserver(private val globalService: GlobalActionBarService?) :
    DefaultLifecycleObserver {
    override fun onStop(owner: LifecycleOwner) {
        globalService?.hideOverlayButton()
    }

    override fun onStart(owner: LifecycleOwner) {
        globalService?.showOverlayButton()
    }
}